doc/run_settings.rst

```diff
@@ -0,0 +1,366 @@
+************
+Run Settings
+************
+
+========
+Overview
+========
+SmartSim supports configuring run settings for entities through the
+``RunSettings`` object which acts as the base class for launching jobs locally or in
+parallel using binaries like mpirun and mpiexec. This ``RunSettings`` base class extends
+to specialized subclasses designed for HPC systems and Workload Managers (WLM) to support
+job launches with environment-specific binaries.
+
+- Navigate to :ref:`Local<Local>` section to configure run settings locally
+- Navigate to :ref:`HPC<HPC>` section to configure run settings for a HPC
+
+A run settings object is initialized through the ``Experiment.create_run_settings()`` function.
+This function accepts a `run_command` argument: the command to run the executable.
+If `run_command` is set to `auto`, SmartSim will attempt to match a run command on the
+system with a ``RunSettings`` class. If found, the class corresponding to
+that `run_command` will be created and returned.
+If the `run_command` is passed a recognized run command (e.g. 'srun') the ``RunSettings``
+instance will be a child class such as ``SrunSettings``.
+If not supported by SmartSim, the base ``RunSettings`` class will be
+created and returned with the specified `run_command` and `run_args` evaluated literally.
+You may also specify `mpirun`, `mpiexec` or `orterun` to the `run_command` argument. This will return
+the associated child class.
+
+Once a ``RunSettings`` object is initialized, you have access to feature set that
+further configures the run settings of a job.
+
+=====
+Local
+=====
+When running SmartSim on laptops and single node workstations, the base ``RunSettings`` object is
+used to parameterize jobs. RunSettings include a `run_command` parameter for local launches
+that utilize a parallel launch binary like `mpirun`, `mpiexec`, and others. If no `run_command`
+is specified, the executable will be launched locally.
+
+The local `launcher` supports the base :ref:`RunSettings API <rs-api>` which can be used to
+run executables as well as run executables with arbitrary launch binaries like `mpirun` and `mpiexec`.
+The local launcher is the default launcher for all ``Experiment`` instances.
+
+When the user initializes the ``Experiment`` at the beginning of the Python driver script,
+a `launcher` argument may be specified. SmartSim will register or detect the `launcher` and return the supported class
+upon a call to ``Experiment.create_run_settings()``. A user may also specify the launcher when initializing
```

Comment:
I took this from the test_run_settings.py file where we pass in the launcher to create_run_settings() - but I agree that I shouldnt expose this param since it will be specified by the user in Exp creation.